### PR TITLE
chore: 19741 numeric validation test add negative amount scenarios

### PR DESCRIPTION
**Pull request by gkozyryatskyy**
**Description**:
Add `Unit tests` with negative values send to HSCS and check the fail results.

**Related issue(s)**:

Fixes [#19741](https://github.com/hiero-ledger/hiero-consensus-node/issues/19741)

**Notes for reviewer**:
- change `getTokenKey(address,uint)` -> `getTokenKey(address,uint256)` is equivalent but keep the same signature styles

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java [hunk 1]
```diff
@@ -27,9 +27,9 @@
  */
 @Singleton
 public class TokenKeyTranslator extends AbstractCallTranslator<HtsCallAttempt> {
-    /** Selector for getTokenKey(address,uint) method. */
+    /** Selector for getTokenKey(address,uint256) method. */
     public static final SystemContractMethod TOKEN_KEY = SystemContractMethod.declare(
-                    "getTokenKey(address,uint)", ReturnTypes.RESPONSE_CODE_TOKEN_KEY)
```
Comment (stoyanov-st) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java]: Why are you changing the types?
https://github.com/hashgraph/hedera-smart-contracts/blob/b243cb6cc87e71526c120b8b2f4e3a7b6fb9c597/contracts/system-contracts/hedera-token-service/HederaTokenService.sol#L610

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x16c/TokenKeyTranslator.java [hunk 2]
```diff
@@ -27,9 +27,9 @@
  */
 @Singleton
 public class TokenKeyTranslator extends AbstractCallTranslator<HtsCallAttempt> {
-    /** Selector for getTokenKey(address,uint) method. */
+    /** Selector for getTokenKey(address,uint256) method. */
     public static final SystemContractMethod TOKEN_KEY_16C = SystemContractMethod.declare(
-                    "getTokenKey(address,uint)", ReturnTypes.RESPONSE_CODE_TOKEN_KEY)
```
Comment (stoyanov-st) [inline hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x16c/TokenKeyTranslator.java]: Same here.

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/ExchangeRateSystemContractTest.java [hunk 3]
```diff
@@ -101,15 +96,15 @@ void inputCannotUnderflow() {
 
         final var result = subject.computeFully(EXCHANGE_RATE_CONTRACT_ID, underflowInput, frame);
         assertThat(result.output()).isEqualTo(Bytes.EMPTY);
-        assertThat(result.result().getHaltReason().get()).isEqualTo(ExceptionalHaltReason.INVALID_OPERATION);
+        assertThat(result.result().getHaltReason().orElse(null)).isEqualTo(ExceptionalHaltReason.INVALID_OPERATION);
```
Comment (lukasz-hashgraph) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/ExchangeRateSystemContractTest.java]: how about `assertThat(result.result().getHaltReason()).hasValue(ExceptionalHaltReason.INVALID_OPERATION);`?

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/ExchangeRateSystemContractTest.java [hunk 4]
```diff
@@ -118,7 +113,7 @@ void selectorMustBeRecognized() {
         final var input = Bytes.concatenate(fragmentSelector, Bytes32.ZERO);
         final var result = subject.computeFully(EXCHANGE_RATE_CONTRACT_ID, input, frame);
         assertThat(result.output()).isEqualTo(Bytes.EMPTY);
-        assertThat(result.result().getHaltReason().get()).isEqualTo(ExceptionalHaltReason.INVALID_OPERATION);
+        assertThat(result.result().getHaltReason().orElse(null)).isEqualTo(ExceptionalHaltReason.INVALID_OPERATION);
```
Comment (lukasz-hashgraph) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/ExchangeRateSystemContractTest.java]: same as above

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java [hunk 5]
```diff
@@ -37,12 +39,12 @@ void returnsUnaliasedOwnerLongZeroForPresentTokenAndNonTreasuryNft() {
                 result.getOutput());
     }
 
-    @Test
-    void returnNonExistingTokenErrorMetadata() {
+    @ParameterizedTest
+    @ValueSource(longs = {-1L, NFT_SERIAL_NO})
+    public void returnErrorMetadata(final long serialNo) {
```
Comment (lukasz-hashgraph) [inline hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java:44]: I'm a bit surprised this completes with `COMPLETED_SUCCESS`, while the other calls revert. Do we know the reason for this inconsistency?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/classiccalls/views/GetTokenKeyFailableCall.java [hunk 6]
```diff
@@ -14,7 +14,7 @@
 
 public class GetTokenKeyFailableCall extends AbstractFailableStaticCall {
 
-    private static final Function SIGNATURE = new Function("getTokenKey(address,uint)");
+    private static final Function SIGNATURE = new Function("getTokenKey(address,uint256)");
```
Comment (stoyanov-st) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/classiccalls/views/GetTokenKeyFailableCall.java]: also here

**Review by stoyanov-st — COMMENTED** [review]
Not sure why this is changing the uint of the keyType.
See the reference linked in the first comment

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java (hunk 5 above)
Comment (gkozyryatskyy) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java:44]: according to implementation `tokenURI(uint256)` return `success` if NFT is not found but return value = `ERC721Metadata: URI query for nonexistent token`. See [TokenUriCall.execute](https://github.com/hiero-ledger/hiero-consensus-node/blob/main/hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenuri/TokenUriCall.java#L49)

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java (hunk 1 above)
Comment (gkozyryatskyy) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java]: from the `Notes for reviewer`
- change `getTokenKey(address,uint)` -> `getTokenKey(address,uint256)` is equivalent but keep the same signature styles
- signature left the same, the goal is to use the same `uint256` type format as in other functions.
- anyway, if you think this is a bad idea, i can rollback

Comment (gkozyryatskyy) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java]: Also, for example in test proxy smart contract we are using uint256. See [here](https://github.com/hiero-ledger/hiero-consensus-node/blob/b4ad4460ecd27d149d9cd6b68c7e0403ec880591/hedera-node/test-clients/src/main/resources/contract/contracts/NumericContract/NumericContract.sol#L99)

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java (hunk 5 above)
Comment (lukasz-hashgraph) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java:44]: Right. To me it would seem slightly better if the erroneous input was handled explicitly with a failure. This would give the caller a more accurate feedback: "your call was invalid, go look for a bug in your code" vs "your call was valid, but this token doesn't exist".

Comment (gkozyryatskyy) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java:44]: There is few points against it:
- We do not really know why it is done like this, but from the code seems it is done on purpose (Because for example call with FT return error)
- This ticket not intended to change the behavior, but to cover existed one

Anyway we can fast discuss this on daily, mb we want to change this. Or mb someone know the reason.

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java (hunk 1 above)
Comment (stoyanov-st) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java]: I'm not stating that the types are not the same. My point is that we must keep in mind the HIP and solidity references in the smart-contracts repo as the source of truth. 
Also the test contract that you referenced is designed for negative test case scenarios so it might have invalid data inside.

Comment (gkozyryatskyy) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java]: Agree that we should have the same sig as in HIP and interface. Will change back.
Also i think, i will change in tests also, to have all in sync

File: hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java (hunk 5 above)
Comment (lukasz-hashgraph) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java:44]: Right, right - it wasn't my intention to change the behavior in this very PR, just something I noticed and thought might be worth a discussion.

Comment (gkozyryatskyy) [reply hedera-node/hedera-smart-contract-service-impl/src/test/java/com/hedera/node/app/service/contract/impl/test/exec/systemcontracts/hts/tokenuri/TokenUriCallTest.java:44]: sure, i understand that =)

File: hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java (hunk 1 above)
Comment (gkozyryatskyy) [reply hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/tokenkey/address_0x167/TokenKeyTranslator.java]: done
